data/secscan_model/secscan_v4_model.py

```diff
@@ -161,3 +200,40 @@ def register_model_cleanup_callbacks(self, data_model_config):
     @property
     def legacy_api_handler(self):
         raise NotImplementedError("Unsupported for this security scanner version")
+
+
+def features_for(report):
```

Comment:
Move the translation layer into its own file

Reply:
Implemented using `namedtuple` subclasses with field validation using `assert` as discussed. Both V2 and V4 instantiate a `SecurityInformation` data container from the API response in their respective `secscan_model` (is based on V2 model, so no transformation needed).